### ISSUE TITLE
Only extracting 20 titles.

### ISSUE THREAD

**Issue by redknob** [joonaspaakko/audible-library-extractor#148]
It won't extract more than 20 titles from my library, which is also the default list length of the library page. I don't have any podcasts. I have 29 titles in my library but no matter what I do, clearing/start again/retry it only ever extracts the first 20.
Changing library view to show 30 titles per page makes no difference.

Tried on Win11 and MacOS with Vivaldi browser using Chrome plugin.

**Comment by joonaspaakko** [comment]
Could you try the dev release and see if that works? I have a feeling it won't, but worth a try.

> 1. [Download the 0.2.11 draft](https://github.com/joonaspaakko/audible-library-extractor/blob/6aefe72c6ad1f5c7d61ee61e196fdb67447e1a5d/audible-library-extractor-0.2.11.zip) (This file sits in the repository root)
> 2. Go to the address: `chrome:extensions` (or: Window > Extensions)
> 3. Turn on the developer mode from the top right
> 4. Drag the downloaded .zip file in the browser window to install

----

Also, just to be sure, could you confirm that you're not getting this issue  #110.

----

To find out more info [open Chrome console](https://developer.chrome.com/docs/devtools/open#shortcuts), run the extraction, and take a screenshot of any errors there.

**Comment by redknob** [comment]
Hi! I can confirm that I am not experiencing what is described in issue #110 .

I have the same issue when using 0.2.11 extension. I managed to install it successfully and disabled the 0.2.10 version first.
Only the first 20 titles are extracted (I have 30 now).

No errors appear during the process.
Re-running the extraction doesn't help either, it still only looks for the first 20.


![image](https://github.com/user-attachments/assets/a54a6826-c002-4781-972a-1cfa5532396e)

![image](https://github.com/user-attachments/assets/0b5c34d6-f030-47b1-8ee0-948f505acdc7)

**Comment by joonaspaakko** [comment]
I can't reproduce the issue (in Vivaldi or Chrome). I'll try to do some more testing. I tested it on Vivaldi. Maybe other extensions might do something that conflicts with it...?  

Maybe I can make a build for you that logs certain key points, and that might tell us why it's happening on your end.

> Note to self: [this](https://github.com/joonaspaakko/audible-library-extractor/blob/6aefe72c6ad1f5c7d61ee61e196fdb67447e1a5d/src/content-script/_components/_mixins/calls/scrapingPrep.js#L76-L77) is part of the code that checks how many pages it has to go through. It's possible that code doesn't work in your situation or maybe it never gets this far and thinks there's only one library page.

**Comment by redknob** [comment]
Hey, I tried it in clean version of Edge as well, same result.
I also tried it in Vivaldi with all other extensions and built-in ad block disabled, same result.

Very strange.

**Comment by joonaspaakko** [comment]
Yea. It's not the first time this happened, but I think that time it happened because Audible changed the layout and it affected everyone, and that's what I expected this time too... I'll try to get you a debug build ASAP (ish). If I don't have time during the week, I'll set Saturday as a hard deadline.

**Comment by joonaspaakko** [comment]
@redknob, I had some trouble with the build but I got you the build [Audible Library Extrator 0.2.11.zip](https://www.dropbox.com/scl/fi/h1qzng6w6fwrtzjutm1q9/audible-library-extractor-0.2.11.zip?rlkey=ul9q2htrzb5lbeyy8ccqalrfn&dl=1) (Couple hours late of the deadline I set for myself. It's am on a sunday...). 

Just to make it clear what to do and what to look out for, [here's a short video](https://share.cleanshot.com/TVSrRhC3). So open browser console, start extraction, wait for the alert, and If you can save the console log or take a screenshot of it or something, that'd be great. I'm not too sure if it'll tell me everything I need to know, I had hard time deciding what to log.

**Comment by redknob** [comment]
Post above is in Vivaldi on MacOS. Tried again in Chrome also on MacOS and only got this far:

![2024-09-05_15-16](https://github.com/user-attachments/assets/bf22a4e1-4fa3-4903-a606-1ab6fa8afdd8)

**Comment by joonaspaakko** [comment]
Interesting how it seems to skip the part that checks how many pages your library has. I suspected that, but I still don't understand why it would. I'll try to take a better look over the weekend.

**Comment by joonaspaakko** [comment]
@redknob, when you extracted 20 books before, did it open the gallery where you could view all 20 of them, or did the extraction just tell you it found 20 books but it never lead to anything?

**Comment by redknob** [comment]
It opened the gallery with 20 books. For the 20 it actually retrieved, everything worksed/looked fine.

**Comment by joonaspaakko** [comment]
When you click book covers in the gallery, can you see the book description? 

> My running theory is that none of your requests go through, except maybe the only reason it's getting books from the first page is that you've already loaded it before you start the extraction.

**Comment by redknob** [comment]
Yep. Just tried it again with the regular version of the extension.
Details pop up on the ALE Library page. All the URLs linking to Audible also work.

**Comment by joonaspaakko** [comment]
@redknob, I've been looking into this a little bit after I got the new release out and I'm still just as baffled as I was before. 

If you're still motivated, could you test the [latest release](https://chromewebstore.google.com/detail/audible-library-extractor/deifcolkciolkllaikijldnjeloeaall)? I was thinking that maybe you should try it in another browser you haven't used before, just to make sure it's not the browser settings or another extension, but you did try at least Vivaldi and Edge so that probably won't change a thing. I'm almost ready to believe you're the only person in Audible with a different looking pagination.

If you feel comfortable with it I think the only realistic way to figure this out would be to have a screensharing session where I think my game plan would be to hand you snippets of code that you feed into the browser console to try and point out why it fails to receive the number of pages in the library and/or send you some builds to try, if it comes to that. We would need at least an hour minimum for this.

**Comment by redknob** [comment]
Hey I had this on my to-do for a while and only got around to it now.

Good news - it seems to work now. All titles are being picked up. 43 out of 43. FYI I'm still using Vivaldi etc, nothing really has changed from the perspective of my browser.

So, I'm not sure what fixed it but good job either way. :) Thanks for looking into it.

**Comment by jefferson136** [comment]
Hey, I’m seeing this same issue on my end.

I’ve tried it in both Arc and Chrome, and also in incognito mode to rule out extensions/session weirdness, but I still can’t get it to extract more than 20 titles. It consistently stops at 20 every time.

Really appreciate all the work you’ve put into the extension and into debugging this issue. Happy to help test anything or provide more details/logs if that would be useful for figuring out what’s going on.

**Comment by joonaspaakko** [comment]
Could you try this build?

> I don't think it will help, because it should fail on one of the very first steps, whereas what that build fixes is kind of in later parts of the extraction.

1. Download the draft [0.2.13.zip from here](https://github.com/joonaspaakko/audible-library-extractor/releases/tag/v.0.2.13-draft).
2. Go to the address: chrome:extensions.
   - Or: Window > Extensions
3. Turn on the developer mode from the top right
4. Drag the downloaded .zip file into the browser window to install
    - You can keep the previous installation, but you should disable it.

If this doesn't help, we may need to schedule like a Teams meeting to figure this out.

**Comment by joonaspaakko** [comment]
Opening this for visibility, since it's not fully resolved.

**Comment by MittchPW** [comment]
Thanks for the link joonaspaakko,

I did update to this 2.13 build and no change.

Unfortunately my issue of missing about half the tags and categories. Most from older titles, but many from the last 50 or so titles all current. I attempted to capture the log, but had an issue, will attempt again. So about 500 or so titles missing the info.

For reference I turned off AdBlock and purged all data for a fresh data pull.

I will attempt to get the log again and upload.

<img width="2368" height="1402" alt="Image" src="https://github.com/user-attachments/assets/db97f816-36c5-4eee-bc46-f94d998c9b20" />

**Comment by joonaspaakko** [comment]
@MittchPW, I was able to replicate this issue, but I haven't been able to figure out what is causing it yet.

**Comment by MittchPW** [comment]
I appreciate you working to find a solution. If you need anything just let me know. Happy to assist if possible.

**Comment by joonaspaakko** [comment]
@MittchPW, can you try it again? I was able to replicate that issue before and didn't really find a good reason for it. Now, I can't seem to be able to replicate it at all.

**Comment by joonaspaakko** [comment]
Closing this for now. Can be reopened if the issue resurfaces.
